IO/Ex1.cpp

```diff
@@ -0,0 +1,216 @@
+
+#include <stdexcept> 
+#include <cassert>
+
+#include "Ex1.hpp"
+
+
+//Ctor
+FileHandler::FileHandler(const std::string path, std::ios_base::openmode permissions)
+: m_path(path), 
+  m_permissions(permissions),
+  m_read_pos(std::streampos(0)) {
+
+    open();
+}
+
+//CCtor, non-const because of tellg/p which are modifying 
+FileHandler::FileHandler(FileHandler &other) 
+: m_path(other.m_path), 
+  m_permissions(other.m_permissions),
+  m_read_pos(other.m_read_pos) {
+
+    open(); 
+
+    auto g = other.m_stream.tellg();
+    auto p = other.m_stream.tellp();
+    if (g == -1 || p == -1) {
+        throw std::runtime_error("CCtor Failed to do tellg/p on file: " + m_path);
+    }
```

Comment:
i see some repetition here.
this check want to answer the question "is the file/stream in a valid state"
are there any other places you want to check this? can you extract to a function?